packages/currency/src/currency-manager.ts

```diff
@@ -52,6 +52,8 @@ export class CurrencyManager<TMeta = unknown> implements ICurrencyManager<TMeta>
     }
     this.legacyTokens = legacyTokens || CurrencyManager.getDefaultLegacyTokens();
     this.conversionPairs = conversionPairs || CurrencyManager.getDefaultConversionPairs();
+    this.chainManager = chainManager || ChainManager.current();
+    ChainManager.setCurrent(this.chainManager);
```

Comment:
ha! this is the trick ;) So this is why it works in request-node.

Reply:
@alexandre-abrioux does that mean `setCurrent` should almost never be called directly?
As calling the constructor to CurrencyManager (or even `getDefault` would override it)

I feel this could lead to unexpected behaviours :thinking_face: